packages/vue-i18n-routing/src/compatibles/utils.ts

```diff
@@ -78,6 +78,29 @@ export function routeToObject(route: Route | RouteLocationNormalizedLoaded) {
   }
 }
 
+/**
+ * This function maps the response of `router.resolve` to properly encode the path.
+ *
+ * @param route - the {@link RouteLocation} provided by `router.resolve`.
+ * @returns a {@link RouteLocation} with URL encoded `fullPath`, `path` and `href` properties.
+ */
+export function resolvedRouteToObject(
+  route: RouteLocation & {
+    href: string
+  }
+): RouteLocation & {
+  href: string
+} {
+  const encodedPath = encodeURI(route.path)
+  const queryString = route.fullPath.indexOf('?') >= 0 ? route.fullPath.substring(route.fullPath.indexOf('?')) : ''
+  return {
+    ...route,
+    fullPath: encodedPath + queryString,
+    path: encodedPath,
+    href: encodedPath + queryString
+  }
+}
+
```

Comment:
The tests for Vue 2 are failing as the route that is being passed could be either a `vue-router` v3 of v4 route. The following change would fix the tests, but if you could manage it while keeping the type annotations it would be preferred 😄

```suggestion
export function resolvedRouteToObject(route: unknown): unknown {
  // use `route.route` for vue-router v3
  // @ts-ignore
  const r = isVue3 ? route : route.route

  const encodedPath = encodeURI(r.path)
  const queryString = r.fullPath.indexOf('?') >= 0 ? r.fullPath.substring(r.fullPath.indexOf('?')) : ''
  const resolvedObject = {
    ...r,
    fullPath: encodedPath + queryString,
    path: encodedPath,
    href: encodedPath + queryString
  }

  // @ts-ignore
  return isVue3 ? resolvedObject : { ...route, route: resolvedObject }
}

```

Reply:
Thanks! Looks like I have bad luck with tests :sweat_smile: 
Your suggestion looks good, I will test it and hopefully manage to add proper typings this weekend

Reply:
After playing around some more, this change would make it typesafe. But don't feel pressured to use my code 😅 I just enjoy the puzzle, and maybe it could be helpful. 💪

```suggestion
type ResolvedRoute = ReturnType<Router['resolve']> | ReturnType<VueRouter['resolve']>['route']
type BridgeRoute = ResolvedRoute | { route: ResolvedRoute }

export function isV4Route(val: BridgeRoute): val is ResolvedRoute {
  return isVue3
}

export function resolveBridgeRoute(val: BridgeRoute) {
  return isV4Route(val) ? val : val.route
}

/**
 * This function maps the response of `router.resolve` to properly encode the path.
 *
 * @param route - the {@link RouteLocation} provided by `router.resolve`.
 * @returns a {@link RouteLocation} with URL encoded `fullPath`, `path` and `href` properties.
 */
export function resolvedRouteToObject(route: BridgeRoute): BridgeRoute {
  const r = resolveBridgeRoute(route)

  const encodedPath = encodeURI(r.path)
  const queryString = r.fullPath.indexOf('?') >= 0 ? r.fullPath.substring(r.fullPath.indexOf('?')) : ''
  const resolvedObject = {
    ...r,
    fullPath: encodedPath + queryString,
    path: encodedPath,
    href: encodedPath + queryString
  }

  return isVue3 ? resolvedObject : { ...route, route: resolvedObject }
}
```

Reply:
Looks great, thanks for your support @BobbieGoede! I just took it as it is :)